# CHIDUBEM IKEDIONWU

I noticed the quiz results had different numbering to the original questions. I left the numbering as was in the questions.

In [1]:
import pandas as pd
import numpy as np

In [2]:
fuel_data = pd.read_csv('https://bit.ly/HDSC-StageOneDataset') 

## QUESTION 3

Which of the following fuel type code has the lowest average fuel cost per unit burned?

In [3]:
fuel_data.groupby('fuel_type_code_pudl')['fuel_cost_per_unit_burned'].sum()/fuel_data['fuel_type_code_pudl'].value_counts()

coal         67.421830
gas          13.659397
nuclear    4955.157002
oil         168.877086
other        18.253856
waste        19.518122
dtype: float64

### ANSWER
Gas 

## QUESTION 4

What is the standard deviation and 75th percentile of the measure of energy per unit (Fuel_mmbtu_per_unit) in two decimal places?

In [4]:
fuel_data['fuel_mmbtu_per_unit'].describe()

count    29523.000000
mean         8.492111
std         10.600220
min          0.000001
25%          1.024000
50%          5.762694
75%         17.006000
max        341.260000
Name: fuel_mmbtu_per_unit, dtype: float64

### ANSWER

10.60 and 17.01

## QUESTION 5

What is the skewness and kurtosis for the fuel quantity burned in two decimal places?

In [5]:
skewness = round(fuel_data['fuel_qty_burned'].skew(),2)
kurtosis = round(fuel_data['fuel_qty_burned'].kurtosis(),2)

print(skewness)
print (kurtosis)

15.85
651.37


### ANSWER

15.85 and 651.37

## QUESTION 6

Which feature has missing values and what is the total number of missing value and percentage of the missing rows as a factor of the total number of rows in three decimal places? 

(Enter answer in the format: Feature: xxx, Total: xxx, Percent: xxx)

In [6]:
fuel_data.isnull().sum()

record_id                         0
utility_id_ferc1                  0
report_year                       0
plant_name_ferc1                  0
fuel_type_code_pudl               0
fuel_unit                       180
fuel_qty_burned                   0
fuel_mmbtu_per_unit               0
fuel_cost_per_unit_burned         0
fuel_cost_per_unit_delivered      0
fuel_cost_per_mmbtu               0
dtype: int64

In [7]:
fuel_data['fuel_unit'].isnull().value_counts(normalize=True)

False    0.993903
True     0.006097
Name: fuel_unit, dtype: float64

### ANSWER
Feature: 'fuel_unit', Total: 180, Percent: 0.610%

## QUESTION 8

In [8]:
fuel_data.corr()

utility_id_ferc1  report_year  fuel_qty_burned  \
utility_id_ferc1                      1.000000     0.093323        -0.057447   
report_year                           0.093323     1.000000         0.012952   
fuel_qty_burned                      -0.057447     0.012952         1.000000   
fuel_mmbtu_per_unit                  -0.066946    -0.110853        -0.080946   
fuel_cost_per_unit_burned            -0.037863     0.013599        -0.018535   
fuel_cost_per_unit_delivered         -0.016414    -0.014043        -0.003551   
fuel_cost_per_mmbtu                   0.006122     0.010261        -0.001896   

                              fuel_mmbtu_per_unit  fuel_cost_per_unit_burned  \
utility_id_ferc1                        -0.066946                  -0.037863   
report_year                             -0.110853                   0.013599   
fuel_qty_burned                         -0.080946                  -0.018535   
fuel_mmbtu_per_unit                      1.000000                  -0.010034   
fuel_cost_per_unit_burned               -0.010034                   1.000000   
fuel_cost_per_unit_delivered            -0.009039                   0.011007   
fuel_cost_per_mmbtu                     -0.005884                  -0.000437   

                              fuel_cost_per_unit_delivered  \
utility_id_ferc1                                 -0.016414   
report_year                                      -0.014043   
fuel_qty_burned                                  -0.003551   
fuel_mmbtu_per_unit                              -0.009039   
fuel_cost_per_unit_burned                         0.011007   
fuel_cost_per_unit_delivered                      1.000000   
fuel_cost_per_mmbtu                              -0.000109   

                              fuel_cost_per_mmbtu  
utility_id_ferc1                         0.006122  
report_year                              0.010261  
fuel_qty_burned                         -0.001896  
fuel_mmbtu_per_unit                     -0.005884  
fuel_cost_per_unit_burned               -0.000437  
fuel_cost_per_unit_delivered            -0.000109  
fuel_cost_per_mmbtu                      1.000000

### ANSWER 

fuel_qty_burned and fuel_mmbtu_per_unit

## QUESTION 9

For the fuel type coal, what is the percentage change in the fuel cost per unit burned in 1998 compared to 1994?

In [9]:
cost_1998 = fuel_data.loc[fuel_data['report_year'] == 1998]['fuel_cost_per_unit_burned'].sum() 
cost_1994 = fuel_data.loc[fuel_data['report_year'] == 1994]['fuel_cost_per_unit_burned'].sum() 

change = (cost_1998 - cost_1994)/cost_1994

percentage_change = change*100

print(percentage_change)


-46.73647708743576


### ANSWER

None of the above.

## QUESTION 10

Which year has the highest average fuel cost per unit delivered?

In [10]:
fuel_data.groupby('report_year')['fuel_cost_per_unit_delivered'].sum()/fuel_data['report_year'].value_counts()

1994       63.636060
1995       32.735269
1996     9196.705948
1997    11140.197239
1998      287.154420
1999       25.551627
2000      985.362877
2001       60.050396
2002       47.594361
2003       55.663493
2004      139.524275
2005       41.438184
2006       38.657484
2007       43.325023
2008       58.588197
2009      652.694163
2010       91.862105
2011       59.774667
2012       60.994502
2013      172.307591
2014      192.737183
2015      326.535511
2016      103.901761
2017       46.196861
2018      499.269966
dtype: float64

### ANSWER

1997